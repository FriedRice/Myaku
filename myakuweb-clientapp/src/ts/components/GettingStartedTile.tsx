/** @module Tile with sample searches component */

import React from 'react';
import Tile from './Tile';

import {
    DEFAULT_SEARCH_OPTIONS,
    Search,
} from '../types';

interface GettingStartedTileProps {
    onSearchSubmit: (search: Search) => void;
}
type Props = GettingStartedTileProps;

interface SampleSearch {
    query: string;
    explanation: string;
}

const SAMPLE_SEARCHES: SampleSearch[] = [
    {
        query: '力士',
        explanation: 'A normal word',
    },
    {
        query: '非常に激しい雨',
        explanation: 'A set phrase',
    },
    {
        query: '急がば回れ',
        explanation: 'A proverb',
    },
    {
        query: '歯が立たない',
        explanation: 'An idiom',
    },
];


function getSearchSubmitHandler(
    query: string, onSearchSubmit: (search: Search) => void
): (event: React.SyntheticEvent) => void {
    return function(event: React.SyntheticEvent): void {
        event.preventDefault();
        onSearchSubmit({
            query: query,
            pageNum: 1,
            options: DEFAULT_SEARCH_OPTIONS,
        });
    };
}

function getSampleSearchLis(
    onSearchSubmit: (search: Search) => void
): React.ReactElement[] {
    var sampleSearchLis: React.ReactElement[] = [];
    for (const sampleSearch of SAMPLE_SEARCHES) {
        sampleSearchLis.push(
            <li key={sampleSearch.query}>
                {`${sampleSearch.explanation} - `}
                <span className='japanese-text' lang='ja'>
                    <a
                        href={`/?q=${sampleSearch.query}`}
                        onClick={getSearchSubmitHandler(
                            sampleSearch.query, onSearchSubmit
                        )}
                    >
                        {sampleSearch.query}
                    </a>
                </span>
            </li>
        );
    }
    return sampleSearchLis;
}

const GettingStartedTile: React.FC<Props> = function(props) {
    return (
        <Tile tileClasses='start-tile'>
            <h4 className='main-tile-header'>Getting Started</h4>
            <p className='list-start-text'>
                Here are some sample searches to get started:
            </p>
            <ul className='myaku-ul'>
                {getSampleSearchLis(props.onSearchSubmit)}
            </ul>
        </Tile>
    );
};

export default GettingStartedTile;
