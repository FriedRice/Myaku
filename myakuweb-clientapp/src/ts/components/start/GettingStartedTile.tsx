/** @module Tile with sample searches component */

import { Link } from 'react-router-dom';
import React from 'react';
import Tile from 'ts/components/generic/Tile';
import { getSearchUrl } from 'ts/app/search';

interface SampleSearch {
    query: string;
    explanation: string;
}

const SAMPLE_SEARCHES: SampleSearch[] = [
    {
        query: '力士',
        explanation: 'A normal word',
    },
    {
        query: '非常に激しい雨',
        explanation: 'A set phrase',
    },
    {
        query: '急がば回れ',
        explanation: 'A proverb',
    },
    {
        query: '歯が立たない',
        explanation: 'An idiom',
    },
];


function getSampleSearchLis(): React.ReactElement[] {
    var sampleSearchLis: React.ReactElement[] = [];
    for (const sampleSearch of SAMPLE_SEARCHES) {
        sampleSearchLis.push(
            <li key={sampleSearch.query}>
                {`${sampleSearch.explanation} - `}
                <span lang='ja'>
                    <Link to={getSearchUrl(sampleSearch.query)}>
                        {sampleSearch.query}
                    </Link>
                </span>
            </li>
        );
    }
    return sampleSearchLis;
}

const GettingStartedTile: React.FC<{}> = function() {
    return (
        <Tile tileClasses='start-tile'>
            <h4 className='main-tile-header'>Getting Started</h4>
            <p className='list-start-text'>
                Here are some sample searches to get started:
            </p>
            <ul className='myaku-ul'>
                {getSampleSearchLis()}
            </ul>
        </Tile>
    );
};

export default GettingStartedTile;
