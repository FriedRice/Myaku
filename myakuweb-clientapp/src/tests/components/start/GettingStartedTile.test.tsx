/**
 * Tests for the GettingStartedTile component.
 * @module tests/components/start/GettingStartedTile.test
 */

import GettingStartedTile from 'ts/components/start/GettingStartedTile';
import React from 'react';
import { shallow } from 'enzyme';


describe('<GettingStartedTile />', function() {
    it('renders correctly', function() {
        const wrapper = shallow(<GettingStartedTile />);
        expect(wrapper).toMatchSnapshot();
    });
});
