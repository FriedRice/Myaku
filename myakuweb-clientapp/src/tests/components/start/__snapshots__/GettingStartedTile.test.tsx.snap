// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GettingStartedTile /> renders correctly 1`] = `
<Tile
  tileClasses="start-tile"
>
  <h4
    className="main-tile-header"
  >
    Getting Started
  </h4>
  <p
    className="list-start-text"
  >
    Here are some sample searches to get started:
  </p>
  <ul
    className="myaku-ul"
  >
    <li
      key="力士"
    >
      A normal word - 
      <span
        lang="ja"
      >
        <Link
          to="/search/?q=力士"
        >
          力士
        </Link>
      </span>
    </li>
    <li
      key="非常に激しい雨"
    >
      A set phrase - 
      <span
        lang="ja"
      >
        <Link
          to="/search/?q=非常に激しい雨"
        >
          非常に激しい雨
        </Link>
      </span>
    </li>
    <li
      key="急がば回れ"
    >
      A proverb - 
      <span
        lang="ja"
      >
        <Link
          to="/search/?q=急がば回れ"
        >
          急がば回れ
        </Link>
      </span>
    </li>
    <li
      key="歯が立たない"
    >
      An idiom - 
      <span
        lang="ja"
      >
        <Link
          to="/search/?q=歯が立たない"
        >
          歯が立たない
        </Link>
      </span>
    </li>
  </ul>
</Tile>
`;
